core/services/pipeline/orm.go

```diff
@@ -271,10 +271,42 @@ func (o *orm) InsertFinishedRun(run *Run, saveSuccessfulTaskRuns bool, qopts ...
 	return errors.Wrap(err, "InsertFinishedRun failed")
 }
 
+// DeleteRunsOlderThan deletes all pipeline_runs that have been finished for a certain threshold to free DB space
 func (o *orm) DeleteRunsOlderThan(ctx context.Context, threshold time.Duration) error {
-	q := o.q.WithOpts(pg.WithParentCtx(ctx))
-	err := q.ExecQ(`DELETE FROM pipeline_runs WHERE finished_at < $1`, time.Now().Add(-threshold))
-	return errors.Wrap(err, "DeleteRunsOlderThan failed")
+	// Addede 1 minute timeout to account for big databases
+	q := o.q.WithOpts(pg.WithParentCtx(ctx), pg.WithLongQueryTimeout())
```

Comment:
Do we have a test on this that deletes at least 1000 runs? Just to double check the batching/sorting

Reply:
I will add one for it